Automattic/wp-calypso
# Hosting Command Palette: Add cache action commands on Calypso

Related to https://github.com/Automattic/dotcom-forge/issues/4723

## Proposed Changes

This PR adds `Clear cache` and `Toggle edge cache` action commands to the command palette. With the `Clear cache` commands, if the global edge cache is not, the command will clear that cache and if it is toggled off, it will clear the WordPress cache which replicates the logic that we have in the edge cache card. 

## Testing Instructions

* Navigate to Calypso Live Link
* Navigate to `/sites`
* Press `cmd + k` on macOS and `ctrl + k` on Windows to trigger command palette
* Search for `Toggle edge cache` command 
* Select the site that you want to use the command on
* Observe that the command is enabling / disabling the toggle (you can cross-check with `hosting-config/{yourSite}` to confirm that the setting changed)

<img width="1522" alt="Screenshot 2023-12-04 at 4 27 03 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/ad7ae5b0-9529-4a41-96aa-14988d407698">

* Try the command and confirm that the settings and the notices change accordingly

**To test Clear cache command**

* Search for `Clear cache` command 
* Select the site on which you want to clear the cache
* Check beforehand if the edge cache is enabled on  `hosting-config/{yourSite}`
* Open the browser's console with the `Network` tab before executing the command 
* Execute the command
* Confirm that if the edge cache is enabled, the requests to clear the cache go to `/hosting/edge-cache/purge`
* Confirm that if the edge cache is enabled, the requests to clear the cache go to `/hosting/clear-cache`

<img width="1523" alt="Screenshot 2023-12-04 at 4 36 15 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/280a7c9a-c7e5-4083-8786-2da3b35a039c">

## Pre-merge Checklist
